connectors/sources/dropbox.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Dropbox source module responsible to fetch documents from Dropbox online.
+"""
+from functools import cached_property
+
+import dropbox
+import requests
+from dropbox.exceptions import ApiError, AuthError, BadInputError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import CancellableSleeps, iso_utc
+
+RETRIES = 3
+
+
+class DropboxClient:
+    """Dropbox client to handle API calls made to Dropbox"""
+
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self.configuration = configuration
+        self.path = (
+            "" if self.configuration["path"] == "/" else self.configuration["path"]
+        )
+        self.include_deleted_files = self.configuration["include_deleted_files"]
+        self.retry_count = self.configuration["retry_count"]
+        self._session = None
+
+    @cached_property
+    def _create_connection(self):
+        self._session = requests.Session()
+        return dropbox.Dropbox(  # pyright: ignore
+            app_key=self.configuration["app_key"],
+            app_secret=self.configuration["app_secret"],
+            oauth2_refresh_token=self.configuration["refresh_token"],
+            max_retries_on_error=self.retry_count,
+            max_retries_on_rate_limit=self.retry_count,
+            session=self._session,
+        )
+
+    def ping(self):
+        self._create_connection.users_get_current_account()
```

Comment:
How do you handle response errors or exceptions?

Reply:
I think we can handle errors in the DataSource's ping method same as we did in `_remote_validation`.